mule-extensions-api/src/main/java/org/mule/runtime/extension/api/annotation/metadata/MetadataKeyPart.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.extension.api.annotation.metadata;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import org.mule.runtime.api.metadata.MetadataKey;
+import org.mule.runtime.api.metadata.resolving.MetadataContentResolver;
+import org.mule.runtime.api.metadata.resolving.MetadataOutputResolver;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+/**
+ * Marks a field in a POJO that represents a composed {@link MetadataKey} as one of the parts
+ * of that composed {@link MetadataKey}.
```

Comment:
we could extablish the relation between the a field of an ObjectType annotated with MetadataKeyId and the MetadataKeyPart. The important thing is that the MetadataKeyPart is a part of the MetadataKey represented by the element annotated with MetadataKeyId
.